### PR TITLE
13159 - Refresh Decks (Do not merge, only for testing build)

### PR DESCRIPTION
**Pull request by claudiociardelli**

### CODE REVIEW

File: vassal-app/src/main/java/VASSAL/counters/Deck.java [hunk 1]
```diff
@@ -1668,4 +1669,67 @@ protected void repaintMap() {
...
   }
+
+  /**
+   * Updates the properties of the Deck in the current open game based on the Deck definition stored in the module (stored in DrawPile.dummy)
+   */
+  public void Refresh() {
+    // Locate the Drawpile in the module that matches the Deck
+    // Update properties
+    for (final DrawPile drawPile : gameModule.getAllDescendantComponentsOf(DrawPile.class)) {
+      if (getDeckName().equals(drawPile.getAttributeValueString(SetupStack.NAME))) {
```
Comment (BrentEaston) [inline vassal-app/src/main/java/VASSAL/counters/Deck.java:1680]: You can just call `DrawPile.findDrawPile(name)` to find a given named DrawPile.

**Review by BrentEaston — CHANGES_REQUESTED** [review]

File: vassal-app/src/main/java/VASSAL/counters/Deck.java (hunk 1 above)
Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/counters/Deck.java:1680]: For that matter why not make the DrawPile a parameter e.g. Refresh(DrawPile drawPile) if at the time you're calling this you already know the drawpile in question?

File: vassal-app/src/main/java/VASSAL/counters/Deck.java [hunk 2]
```diff
@@ -1668,4 +1669,67 @@ protected void repaintMap() {
       map.repaint();
     }
   }
+
+  /**
+   * Updates the properties of the Deck in the current open game based on the Deck definition stored in the module (stored in DrawPile.dummy)
+   */
+  public void Refresh() {
```
Comment (Cattlesquat) [inline vassal-app/src/main/java/VASSAL/counters/Deck.java:1676]: Usually first letter in a method isn't capitalized?

**Review by Cattlesquat — COMMENTED** [review]
Can we get this finished and in for 3.6?

**Comment by riverwanderer** [comment]
> Can we get this finished and in for 3.6?
I'm happy to help with any testing.

**Comment by Cattlesquat** [comment]
Okay - it seems like Claudio may have hit a busy patch in-real-life, so I may steal things from this and make a new PR.

**Comment by claudiociardelli** [comment]
Hello Brian,
Yes thanks, if you can take this up. I am too busy in RL for the moment to focus on this topic.
Claudio

**Comment by Cattlesquat** [comment]
Replaced by PR #10719

**Pull request closed without merging**